### ISSUE TITLE
Classes should not automatically inherit function signatures from interfaces

### ISSUE THREAD

**Issue by muglug** [vimeo/psalm#336]
This currently isn't caught by Psalm - it doesn't complain about `A::foo` missing a return type, because it supposes it's inherited from `I1`. But `I2` also has a matching docblock.

```php
interface I1 {
  /** @return string */
  public function foo();
}
interface I2 {
  /** @return int */
  public function foo();
}
class A implements I1, I2 {
  public function foo() {
    return "hello";
  }
}
```

This is a fatal error in PHP, not caught by Psalm

```php
interface I1 {
  public function foo() : string;
}
interface I2 {
  public function foo() : int;
}
class A implements I1, I2 {
  public function foo() : string {
    return "hello";
  }
}
```
